snapcraft/cli/assertions.py

```diff
@@ -204,6 +229,96 @@ def list_validation_sets(name, sequence):
         )
 
 
+@assertionscli.command("edit-validation-sets")
+@click.argument("account-id", metavar="<account-id>")
+@click.argument("set-name", metavar="<set-name>")
+@click.argument("sequence", metavar="<sequence>", type=int)
+@click.option("--key-name", metavar="<key-name>")
+def edit_validation_sets(account_id: str, set_name: str, sequence: int, key_name: str):
+    """Edit the list of validations for <set-name>.
+
+    Refer to https://snapcraft.io/docs/validation-sets for further information
+    on Validation Sets.
+    """
+    store_client = StoreClientCLI()
+
+    asserted_validation_sets = store_client.get_validation_sets(
+        name=set_name, sequence=str(sequence)
+    )
+
+    try:
+        # assertions should only have one item since a specific
+        # sequence was requested.
+        revision = asserted_validation_sets.assertions[0].revision
+        snaps = yaml_utils.dump(
+            {
+                "snaps": [
+                    s.marshal() for s in asserted_validation_sets.assertions[0].snaps
+                ]
+            }
+        )
+    except IndexError:
+        # If there is no assertion for a given sequence, the store API
+        # will return an empty list.
+        revision = "0"
+        snaps = _VALIDATIONS_SETS_SNAPS_TEMPLATE
+
+    unverified_validation_sets = _VALIDATION_SETS_TEMPLATE.format(
+        account_id=account_id,
+        set_name=set_name,
+        sequence=sequence,
+        revision=revision,
+        snaps=snaps,
+    )
+
+    edited_validation_sets = _edit_validation_sets(unverified_validation_sets)
+    if edited_validation_sets == yaml_utils.load(unverified_validation_sets):
+        echo.warning("No changes made.")
+    else:
+        build_assertion = store_client.post_validation_sets_build_assertion(
+            validation_sets=edited_validation_sets
+        )
+        signed_validation_sets = _sign_assertion(
+            build_assertion.marshal(), key_name=key_name
+        )
+        store_client.post_validation_sets(signed_validation_sets=signed_validation_sets)
+
+
+def _edit_validation_sets(validation_sets: str) -> Dict[str, Any]:
+    """Spawn an editor to modify the validation-sets."""
+    editor_cmd = os.getenv("EDITOR", "vi")
+
+    with tempfile.NamedTemporaryFile() as ft:
+        ft.close()
```

Comment:
Perhaps just write to it then close it.

Reply:
I don't understand this comment, but we can iterate on this one if you want